cpp/src/Slice/Grammar.y

```diff
@@ -1722,14 +1683,6 @@ metadata_list
 
     $$ = metadataList;
 }
-| metadata_list ',' BAD_TOKEN
-{
-    $$ = $1;
-}
-| BAD_TOKEN
-{
-    $$ = make_shared<MetadataListTok>();
-}
```

Comment:
I removed this rule, because otherwise Bison will 'expect' to see a BAD_TOKEN and say this to users:
`syntax error: unexpected ..., expected string literal or invalid character`
And we should never advise users to put invalid characters in their slice.

After I removed this rule it just says:
`syntax error: unexpected ..., expected string literal`

----

For the record, this rule was about detecting unquoted strings in metadata (ZeroC Slice requires all metadata be quoted). But now the parser just treats them as-if they were quoted (but we still emit an error of course).